File: concrete-testing/src/main/java/com/jaynewstrom/concretetesting/ContextChildTestBlock.java
package com.jaynewstrom.concretetesting;

import com.jaynewstrom.concrete.ConcreteBlock;

final class ContextChildTestBlock implements ConcreteBlock<ContextChildTestComponent> {
    private final ContextTestComponent contextTestComponent;

    ContextChildTestBlock(ContextTestComponent contextTestComponent) {
        this.contextTestComponent = contextTestComponent;
    }

    @Override public String name() {
        return getClass().getName();
    }

    @Override public ContextChildTestComponent createComponent() {
        return DaggerContextChildTestComponent.builder().contextTestComponent(contextTestComponent).build();
    }
}

File: concrete-testing/src/main/java/com/jaynewstrom/concretetesting/ContextChildTestBlock.kt
package com.jaynewstrom.concretetesting

import com.jaynewstrom.concrete.ConcreteBlock

internal class ContextChildTestBlock(
    private val contextTestComponent: ContextTestComponent
) : ConcreteBlock<ContextChildTestComponent> {
    override fun name(): String = javaClass.name

    override fun createComponent(): ContextChildTestComponent {
        return DaggerContextChildTestComponent.builder().contextTestComponent(contextTestComponent).build()
    }
}
